sbi/neural_nets/__init__.py

```diff
@@ -1,26 +1,36 @@
-from sbi.neural_nets.classifier import (
-    build_linear_classifier,
-    build_mlp_classifier,
-    build_resnet_classifier,
-)
-from sbi.neural_nets.embedding_nets import (
-    CNNEmbedding,
-    FCEmbedding,
-    PermutationInvariantEmbedding,
-)
-from sbi.neural_nets.estimators import ConditionalDensityEstimator, NFlowsFlow
 from sbi.neural_nets.factory import (
     classifier_nn,
     flowmatching_nn,
     likelihood_nn,
     posterior_nn,
+    posterior_score_nn,
 )
-from sbi.neural_nets.flow import (
-    build_made,
-    build_maf,
-    build_maf_rqs,
-    build_nsf,
-    build_zuko_maf,
-)
-from sbi.neural_nets.mdn import build_mdn
-from sbi.neural_nets.mnle import MixedDensityEstimator, build_mnle
+
+class CNNEmbedding():
+    """Exists only to raise an explicit error for imports of embedding networks."""
+    def __init__(self, *args, **kwargs):
+        raise ValueError(
```

Comment:
This will be raised only during runtime, right? 
Isn't there a way to raise this at import time?

Reply:
I tried for a while and also quizzed ChatGPT, but could not find a solution that allows some things to be imported (e.g. `posterior_nn`) and prohibits imports of others.

Reply:
Okay I quizzed it a bit more and found sth that works :) Will commit it